sys/xtimer/xtimer_core.c

```diff
@@ -154,7 +153,8 @@ static bool _timer_comparison(xtimer_t* timerA, xtimer_t* timerB)
         return true;
     }
     if (timerA->long_offset == timerB->long_offset
-        && timerA->start_time + timerA->offset - timerB->start_time <= timerB->offset) {
+        && (timerA->start_time + timerA->offset < timerB->start_time
```

Comment:
Added this condition for proper timer comparison. Without this line, timer list is not ordered correctly sometimes.